assets/src/components/form/select/model-selects/status.js

```diff
@@ -0,0 +1,60 @@
+/**
+ * Internal imports
+ */
+import ModelSelect from '../model-select';
+import * as statusModel from '../../../../data/model/status';
+
+/**
+ * External imports
+ */
+import { __ } from '@eventespresso/i18n';
+import { Component } from '@wordpress/element';
+import { PropTypes } from 'prop-types';
+
+/**
+ * Select Component for the Status Model.
+ */
+export default class StatusSelect extends Component {
+	state = {
+		modelName: 'status',
+	};
+
+	static defaultProps = {
+		selectConfiguration: {
+			loadingMessage: () => __( 'Retrieving Statuses.', 'event_espresso' ),
+			noOptionsMessage: () => __( 'No Statuses.', 'event_espresso' ),
+			placeholder: __( 'Select Status...', 'event_espresso' ),
+		},
+		...statusModel.defaultQueryData,
+		getQueryString: statusModel.getQueryString,
+		selectedType: 'registration',
+		selectLabel: __( 'Select Status', 'event_espresso' ),
+		addAllOptionLabel: __( 'All Statuses', 'event_espresso' ),
+	};
+
+	static propTypes = {
+		...statusModel.queryDataTypes,
+		selectedType: PropTypes.string.isRequired,
```

Comment:
Since this has a default set, the `isRequired` usage here is redundant.  If you want to use `isRequired` then remove the default (or leave the default and remove `isRequired`).

Reply:
hmmm... think I'll remove the default then and leave the component more abstracted